libs/vm_modules/tests/unit/state_tests.cpp

```diff
@@ -20,8 +20,7 @@
 
 #include "gmock/gmock.h"
 
-namespace fetch {
```

Comment:
Just comment: I would not drop the namespaces here. Namespaces were here intentionally. The idea behind this is to set unit test code in to the very same namespace as tested production code is implemented in, in order to have the very same setup & experience (access to types) as the code would be written in tested production code. This helps to identify types which are missing (code won't compile) what it might indicate that there is potential unexpected dependency.

The most inner-nested _anonymous_ namespace is here intentionally as well, in order to avoid potential/accidental leak of types declared & defined by unit test code in to production code by linking. This can happen **IF** there is miss-configuration in build setup with unintended consequences which make production code to link with unit test objects.

Reply:
As for setting the namespaces to provide the same 'environment' in terms of the available symbols as the production code, the using namespace declaration I introduced achieves the same goal.

The anonymous namespace is still there. The need for internal linkage in this case is clear.

I think these namespaces should be removed, because all they achieve is making the classes defined under the hood by `TEST_F` to think that they belong in a `fetch` namespace. I do not think hey do, as these classes are code defined by gtest.

Reply:
:+1: